analyzer/evmtokenbalances/evm_token_balances.go

```diff
@@ -146,25 +155,57 @@ func (m Main) getStaleTokenBalances(ctx context.Context, limit int) ([]*StaleTok
 }
 
 func (m Main) processStaleTokenBalance(ctx context.Context, batch *storage.QueryBatch, staleTokenBalance *StaleTokenBalance) error {
-	m.logger.Info("downloading", "stale_token_balance", staleTokenBalance)
-	// todo: assert that token addr and account addr contexts are secp256k1
-	tokenEthAddr, err := client.EVMEthAddrFromPreimage(staleTokenBalance.TokenAddrContextIdentifier, staleTokenBalance.TokenAddrContextVersion, staleTokenBalance.TokenAddrData)
-	if err != nil {
-		return fmt.Errorf("token address: %w", err)
-	}
 	accountEthAddr, err := client.EVMEthAddrFromPreimage(staleTokenBalance.AccountAddrContextIdentifier, staleTokenBalance.AccountAddrContextVersion, staleTokenBalance.AccountAddrData)
 	if err != nil {
 		return fmt.Errorf("account address: %w", err)
 	}
-	if staleTokenBalance.Type != nil {
-		balanceData, err := runtime.EVMDownloadTokenBalance(
+	switch staleTokenBalance.Type {
+	case evm.EVMTokenTypeUnsupported:
+		// Do nothing; we'll just mark this token as processed so we remove it from the queue.
+	case evm.EVMTokenTypeNative:
+		// Query native balance.
+		addr := nodeapi.Address{}
+		if err := addr.UnmarshalText([]byte(staleTokenBalance.AccountAddr)); err != nil {
+			m.logger.Error("invalid account address bech32 '%s': %w", staleTokenBalance.AccountAddr, err)
+			break
```

Comment:
Should this be a `return err`?

Reply:
In practice it probably doesn't matter because we're very unlikely to hit this case.

In theory, we want this to behave much like `case evm.EVMTokenTypeUnsupported:` -- this is a "weird" token-account pair that we aren't ever able to process, but we still want to mark them as "processed" (rather, "looked at") at this height so that they get removed from the queue (implemented in the DB as an index) so it doesn't grow without bounds. It's a somewhat brittle setup :/ and incurs some busywork, but it's fine for now.

I added a comment inline to clarify.

Reply:
Ok thanks!